### DISCUSSION TITLE
Running as a service (uwsgi) not reading environment variables

### DISCUSSION THREAD

**Discussion by travisjohnsonga** [nautobot/nautobot#4654] [General]
Hi all - 

working through https://github.com/nautobot/nautobot-plugin-golden-config/discussions/657

When I run via 'nautobot-server runserver 0.0.0.0 8080 --insecure'  everything is fine.

When I run as a service, none of the envars are picked up, which is causing the secrets for git repo in the config backup (and probably other things) to not work.

my uwsgi and systemd daemon files are correct per the install guides.

**Comment by gsnider2195** [comment]
How are you defining your environment variables? They should all be defined in the systemd service file:

```
[Service]
Type=simple
Environment="NAUTOBOT_ROOT=/opt/nautobot"
Environment="NAUTOBOT_JOBS_ROOT=/source/local_jobs"
```

**Reply by jvanderaa** [reply]
Does this post help: https://josh-v.com/nautobot-environment-file/
